packaging/technical-addon/packaging-scripts/cicd-tests/happypath-test.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash -eux
+
+set -o pipefail
+which jq || (echo "jq not found" && exit 1)
+source "${SOURCE_DIR}/packaging-scripts/cicd-tests/add-access-token.sh"
+BUILD_DIR="$(realpath "$BUILD_DIR")"
+TA_FULLPATH="$(repack_with_access_token "$OLLY_ACCESS_TOKEN" "$BUILD_DIR/out/distribution/Splunk_TA_otel.tgz" | tail -n 1)"
+CI_JOB_ID="${CI_JOB_ID:-$(basename $(dirname "$TA_FULLPATH"))}"
+TEST_FOLDER="${TEST_FOLDER:-$BUILD_DIR/$CI_JOB_ID}"
+mkdir -p "$TEST_FOLDER"
+
+# Create ORCA container & grab id
+splunk_orca -vvv --cloud ${ORCA_CLOUD} --printer sdd-json --deployment-file "$TEST_FOLDER/orca_deployment.json" --ansible-log ansible-local.log create --prefix "happypath" --env SPLUNK_CONNECTION_TIMEOUT=600 --platform $SPLUNK_PLATFORM --splunk-version "${UF_VERSION}" --local-apps "$TA_FULLPATH" --playbook "$SOURCE_DIR/packaging-scripts/orca-playbook-$PLATFORM.yml,site.yml"
+# TODO use jq not awk
+deployment_id="$(grep "orca_deployment_id" "$TEST_FOLDER/orca_deployment.json" | awk -F ':' '{print $2}' | awk -F '"' '{print $2}')"
+echo "$deployment_id" > "$TEST_FOLDER/deployment_id.txt"
+ip_addr="$(jq -r '.server_roles.standalone[0].host' < "$TEST_FOLDER/orca_deployment.json")"
+
+if [ "$PLATFORM" == "windows" ]; then
+    # Windows takes forever to extract
+    echo "sleeping for 700s at $(date)"
+    sleep 700s
+else
+    # Can likely drop this way down, but give the collector time to collect metrics/traces
+    echo "sleeping for 90s at $(date)"
+    sleep 90s
+fi
+
+
+# Copy logs from container
+scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -r -i ~/.orca/id_rsa "splunk@$ip_addr:/opt/splunk/var/log/splunk/" "$TEST_FOLDER"
+
+# Verify Otel agent is running without any error
+grep -q "Starting otel agent" "$TEST_FOLDER/splunk/Splunk_TA_otel.log"
+grep -q "Everything is ready" "$TEST_FOLDER/splunk/otel.log"
+(grep -qi "ERROR" "$TEST_FOLDER/splunk/Splunk_TA_otel.log" && exit 1 ) || true
+(grep -qi "ERROR" "$TEST_FOLDER/splunk/otel.log" && exit 1 ) || true
+
+# Verify Olly has received metrics data from this host
+MAX_ATTEMPTS=6
+DELAY=10
+ATTEMPT=1
+CUTOFF="$(date '+%s%3N' -d '5 min ago')"
+export CUTOFF
+otel_hostname="$(grep "host.name" "$TEST_FOLDER/splunk/otel.log" | head -1 | awk -F 'host.name":"' '{print $2}' | awk -F '","' '{print $1}')"
+while [ $ATTEMPT -le $MAX_ATTEMPTS ]; do
+    curl --header "Content-Type:application/json" --header "X-SF-TOKEN:${OLLY_ACCESS_TOKEN}" "https://api.us0.signalfx.com/v2/metrictimeseries?query=host.name:${otel_hostname}%20AND%20sf_metric:otelcol_process_uptime%20AND%20splunk.distribution:otel-ta" > "$TEST_FOLDER/uptime.json"
+    count=$( grep '"count"' "$TEST_FOLDER/uptime.json" | awk -F ':\ ' '{print $2}' | awk -F ',' '{print $1}')
+
+    if [[ "$count" -gt "0" ]] && jq '[.results[].created, .results[].lastUpdated] | max as $max | $max >= ($ENV.CUTOFF | tonumber)' "$TEST_FOLDER/uptime.json" ; then
+        break
+    else
+        ATTEMPT=$((ATTEMPT + 1))
+        sleep $DELAY
```

Comment:
Maybe take this out of else, since `if` breaks? Feel free to drop, not a big deal.